### PR TITLE
test(utils): added dates

### LINKED ISSUE #79: Dates – Test date manipulations and edge cases (Easy)

### PR DESCRIPTION
**Pull request by tirirote**
Closes #79

### CODE REVIEW

File: src/utils/dates-original.js
Comment (alvarobernal2412) [inline src/utils/dates-original.js:1]: What's this file? This should be removed

File: src/utils/checkRequiredProperties.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ export function checkRequiredProperties(obj, requiredProps) {
   if (!obj || typeof obj !== 'object') {
     return { validation: false, textError: 'Invalid object or missing required properties' };
   }
+
```
Comment (alvarobernal2412) [inline src/utils/checkRequiredProperties.js]: You should avoid committing this, try to touch only the files directly related to the issue.

File: tests/unit/utils/pruebaDates.js
Comment (alvarobernal2412) [inline tests/unit/utils/pruebaDates.js:1]: This file should not be committed

**Review by alvarobernal2412 — CHANGES_REQUESTED** [review]
Here are some improvements you can make to your unit tests for `getDates`:  

### **1. Add Edge Case Tests**  
- **Empty Range:** Test when `from` and `to` dates are the same.  
- **Reverse Order:** Pass a `from` date that is after the `to` date.  
- **Leap Year Handling:** Test February dates, especially for leap years.  
- **Timezone Considerations:** Check if timezone differences affect results.  
- **Start at Non-Boundary Points:** Test cases where the start date isn't aligned with the expected period (e.g., start on Jan 15 for a monthly interval).  

### **2. Use `parseISO` for Consistency**  
Instead of `new Date('YYYY-MM-DD')`, use `parseISO('YYYY-MM-DD')` to avoid potential timezone shifts in some environments.  

### **3. Verify Array Length**  
Instead of just checking equality, also confirm that the expected number of dates is correct:  
```js
expect(result.length).toBe(expectedDates.length);
```

### **4. Validate Output Type**  
Ensure each item in `result` is a valid `Date` object:  
```js
result.forEach(date => expect(date).toBeInstanceOf(Date));
```

### **5. Parameterize Repetitive Tests**  
Avoid duplicating similar test logic for different intervals by using `describe.each`:  
```js
describe.each([
  ['yearly', '2020-01-01', '2024-01-01', ['2020-01-01', '2021-01-01', '2022-01-01', '2023-01-01', '2024-01-01']],
  ['monthly', '2023-01-01', '2023-03-01', ['2023-01-01', '2023-02-01', '2023-03-01']],
])('Testing getDates for %s period', (period, from, to, expected) => {
  it(`should generate correct ${period} dates`, () => {
    const result = getDates(parseISO(from), parseISO(to), period);
    expect(result).toEqual(expected.map(parseISO));
  });
});
```
This reduces redundancy while keeping tests clear.  

### **6. Test Invalid Inputs**  
Check how `getDates` handles:  
- **Null/undefined inputs**  
- **Invalid date strings**  
- **Unsupported period types** (e.g., `getDates(from, to, 'decadely')`)

File: src/utils/checkRequiredProperties.js (hunk 1 above)
Comment (tirirote) [reply src/utils/checkRequiredProperties.js]: this is fixed

File: src/utils/dates-original.js
Comment (tirirote) [reply src/utils/dates-original.js:1]: fixed already

File: tests/unit/utils/pruebaDates.js
Comment (tirirote) [reply tests/unit/utils/pruebaDates.js:1]: fixed too

**Review by alvarobernal2412 — CHANGES_REQUESTED** [review]
The changes requested above have not been resolved. In addition, these lines of code have not been tested: 
Captura de pantalla 2025-04-08 a las 15 06 06
Captura de pantalla 2025-04-08 a las 15 06 09
Captura de pantalla 2025-04-08 a las 15 06 03

>[!CAUTION]
> To reproduce this, run `npm run test:ui`*

**Review by alvarobernal2412 — APPROVED** [review]
